frontend/src/components/RequestDialog.vue

```diff
@@ -132,19 +96,19 @@ const submitRequestTransaction = async (formResult: RequestFormResult) => {
   );
 };
 
-watch(chainId, () => location.reload());
-
-const shownError = computed(() => {
-  return requestSignerError.value || transferError.value;
+watch(chainId, (oldChainId) => {
+  if (oldChainId === -1) {
```

Comment:
This causes the same issue I'm having in my WalletConnect PR at the moment and couldn't find a way to resolve yet! If you are trying to change between Boba and Metis after initial screen loading, page then is not reloading and no Token appears in select box!

Reply:
Good catch. The reason is very simple and stupid. The signature of a `WatchCallback` is `(newValue, oldValue) => void`. So I need to put it `(_, oldChainId)` and then `if (oldChainId !== -1)`. Pushing a fixup commit.

Reply:
Please confirm that it works for you.

Reply:
Yup, it fixes a part of the issue, but not the issue itself. So as per our chat, we can figure out a solution for these doomed watchers!

Reply:
Okay, the problem is a bit more complex. On it...

Reply:
Alright. That should be fixed now by [f8b9eff](https://github.com/beamer-bridge/beamer/pull/668/commits/f8b9effa6002500173bcb79c640fe49db680f14a). Please check it out. 

I guess due to that it makes sense to merge this PR soon, so you can rebase your #647 on top of it to benefit from this fix too.

Reply:
Amazing.. Works perfectly! Thank you!